### PR TITLE
feat(core): move request logging from client to a transport

### LINKED ISSUE #637: Refactor logging of http requests to use an http.Transport
Refactor the logging in the SDK to use Transport for the logging.

Roundtripper can be used as a chained decorator upon an HTTP request that can implement logging, authentication management, header manipulation, ...
Beware to chain them in the right order to avoid a logging transport to leak credentials.

### PR DESCRIPTION
**Pull request by Codelax**
Fix #637

### CODE REVIEW

File: internal/auth/token.go [hunk 1]
```diff
@@ -25,12 +25,18 @@ func (t *Token) Headers() http.Header {
 	return headers
 }
 
+func AnonymizeHeaders(headers http.Header) http.Header {
```
Comment (Monitob) [inline internal/auth/token.go:28]: Can you rename this function may anonymizeTokenHeader

File: internal/auth/token.go [hunk 2]
```diff
@@ -25,12 +25,18 @@ func (t *Token) Headers() http.Header {
...
+	key := headers.Get(XAuthTokenHeader)
+	if key != "" {
+		headers.Set(XAuthTokenHeader, HideSecretKey(key))
+	}
+	return headers
+}
+
 // AnonymizedHeaders returns an anonymized version of Headers()
 // This method could be use for logging purpose.
 func (t *Token) AnonymizedHeaders() http.Header {
```
Comment (Monitob) [inline internal/auth/token.go:38]:
```suggestion
func (t *Token) AnonymizeHeaders() http.Header {
```

File: internal/auth/token.go [hunk 3]
```diff
@@ -25,12 +25,18 @@ func (t *Token) Headers() http.Header {
...
 
+func AnonymizeHeaders(headers http.Header) http.Header {
+	key := headers.Get(XAuthTokenHeader)
+	if key != "" {
+		headers.Set(XAuthTokenHeader, HideSecretKey(key))
+	}
+	return headers
+}
+
 // AnonymizedHeaders returns an anonymized version of Headers()
```
Comment (Monitob) [inline internal/auth/token.go:36]:
```suggestion
// AnonymizeHeaders returns an anonymized version of Headers()
```

File: internal/auth/token.go (hunk 2 above)
Comment (Codelax) [reply internal/auth/token.go:38]: this return the headers of the the token but anonymiz**ed**

**Review by Monitob — APPROVED** [review]

**Pull request merged**